crates/sui-indexer/src/apis/read_api.rs

```diff
@@ -169,15 +169,21 @@ where
             .await
     }
 
-    async fn get_object(&self, object_id: ObjectID) -> RpcResult<GetObjectDataResponse> {
-        self.fullnode.get_object(object_id).await
+    async fn get_object_with_options(
+        &self,
+        object_id: ObjectID,
+        options: Option<SuiObjectContentOptions>,
+    ) -> RpcResult<SuiObjectWithStatus> {
+        self.fullnode
+            .get_object_with_options(object_id, options)
+            .await
     }
 
     async fn get_dynamic_field_object(
         &self,
         parent_object_id: ObjectID,
         name: DynamicFieldName,
-    ) -> RpcResult<GetObjectDataResponse> {
+    ) -> RpcResult<SuiObjectWithStatus> {
```

Comment:
nit: we can call this SuiObjectResponse? (consistent with existing `SuiTransactionResponse`)